### ISSUE TITLE
data race when opening/closing a connection

### ISSUE THREAD

**Issue by jxsl13** [jxsl13/amqpx#10]
```text
WARNING: DATA RACE
Write at 0x00c000214050 by goroutine 1422:
  github.com/rabbitmq/amqp091-go.(*Connection).openComplete()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:909 +0x150
  github.com/rabbitmq/amqp091-go.(*Connection).openVhost()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:894 +0x1c9
  github.com/rabbitmq/amqp091-go.(*Connection).openTune()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:880 +0xc66
  github.com/rabbitmq/amqp091-go.(*Connection).openStart()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:813 +0x66f
  github.com/rabbitmq/amqp091-go.(*Connection).open()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:785 +0xc4
  github.com/rabbitmq/amqp091-go.Open()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:266 +0x664
  github.com/rabbitmq/amqp091-go.DialConfig()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:246 +0x904
  github.com/jxsl13/amqpx/pool.(*Connection).connect()
      /home/behm015/Development/amqpx/pool/connection.go:168 +0x33a
  github.com/jxsl13/amqpx/pool.(*Connection).Connect()
      /home/behm015/Development/amqpx/pool/connection.go:156 +0x90
  github.com/jxsl13/amqpx/pool.NewConnection()
      /home/behm015/Development/amqpx/pool/connection.go:97 +0x929
  github.com/jxsl13/amqpx/pool_test.TestNewConnectionDisconnect.func1()
      /home/behm015/Development/amqpx/pool/connection_test.go:59 +0x26b
  github.com/jxsl13/amqpx/pool_test.TestNewConnectionDisconnect.func2()
      /home/behm015/Development/amqpx/pool/connection_test.go:77 +0x47

Previous write at 0x00c000214050 by goroutine 1599:
  github.com/rabbitmq/amqp091-go.(*Connection).shutdown.func1()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:489 +0x511
  sync.(*Once).doSlow()
      /home/behm015/sdk/go1.19/src/sync/once.go:74 +0x101
  sync.(*Once).Do()
      /home/behm015/sdk/go1.19/src/sync/once.go:65 +0x46
  github.com/rabbitmq/amqp091-go.(*Connection).shutdown()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:453 +0x84
  github.com/rabbitmq/amqp091-go.(*Connection).reader()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:596 +0x444
  github.com/rabbitmq/amqp091-go.Open.func1()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:265 +0x58

Goroutine 1422 (running) created at:
  github.com/jxsl13/amqpx/pool_test.TestNewConnectionDisconnect()
      /home/behm015/Development/amqpx/pool/connection_test.go:56 +0xee
  testing.tRunner()
      /home/behm015/sdk/go1.19/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /home/behm015/sdk/go1.19/src/testing/testing.go:1493 +0x47

Goroutine 1599 (finished) created at:
  github.com/rabbitmq/amqp091-go.Open()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:265 +0x629
  github.com/rabbitmq/amqp091-go.DialConfig()
      /home/behm015/go/pkg/mod/github.com/rabbitmq/amqp091-go@v1.5.0/connection.go:246 +0x904
  github.com/jxsl13/amqpx/pool.(*Connection).connect()
      /home/behm015/Development/amqpx/pool/connection.go:168 +0x33a
  github.com/jxsl13/amqpx/pool.(*Connection).Connect()
      /home/behm015/Development/amqpx/pool/connection.go:156 +0x90
  github.com/jxsl13/amqpx/pool.NewConnection()
      /home/behm015/Development/amqpx/pool/connection.go:97 +0x929
  github.com/jxsl13/amqpx/pool_test.TestNewConnectionDisconnect.func1()
      /home/behm015/Development/amqpx/pool/connection_test.go:59 +0x26b
  github.com/jxsl13/amqpx/pool_test.TestNewConnectionDisconnect.func2()
      /home/behm015/Development/amqpx/pool/connection_test.go:77 +0x47
==================
--- FAIL: TestNewConnectionDisconnect (16.12s)
```

**Comment by jxsl13** [comment]
Opened an issue at the underlying amqp library
https://github.com/rabbitmq/amqp091-go/issues/170

**Comment by jxsl13** [comment]
fixed in v0.6.1
